### DISCUSSION TITLE
How to use previous AWS query protocol in Rust AWS SQS Client instead of JSON protocol?

### DISCUSSION THREAD

**Discussion by AlexanderSerbul** [awslabs/aws-sdk-rust#1156] [Q&A] [answered]
Good day!

By default, AWS SDK for Rust SQS Client uses JSON query to SQS queue. (https://docs.aws.amazon.com/AWSSimpleQueueService/latest/APIReference/API_GetQueueUrl.html). Is it possible to switch client to use previous "query protocol"?

I use such code and, by default, it uses JSON protocol t speak to AWS SQS:
```
async fn yc_sqs_client(c: Config) -> aws_sdk_sqs::Client {
    let sdk_config = aws_config::defaults(BehaviorVersion::v2024_03_28())
        .region(aws_config::Region::new(c.yc_region))
        .profile_name(c.yc_profile)
        .load()
        .await;

    let sqs_local_config = aws_sdk_sqs::config::Builder::from(&sdk_config)
        .endpoint_url(c.yc_sqs_endpoint)
        .build();

    aws_sdk_sqs::Client::from_conf(sqs_local_config)

...

        let c = Config::get_with_defaults(true);

        let qurl = yc_sqs_client(c.clone()).await.get_queue_url()
            .queue_name("yc-sqs").send().await.unwrap().queue_url;

        assert_eq!(qurl.unwrap(), "https://***");
```

Some cloud providers not yet support SQS API with JSON protocol and we need to use them.

https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-json-faqs.html
https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-release-notes.html

**Comment by aajtodd** [comment, marked as answer]
Thanks for the question. Unfortunately, no there is not a way to revert this to use the previous protocol. Smithy client generators, including `smithy-rs`, select a single protocol to generate for. We have no plans to generate and publish both versions and can only assert correctness and compatibility with the actual AWS SQS service. You'll need to wait for the upstream "SQS like" API to be updated.
